bot/cogs/temporary.py

```diff
@@ -0,0 +1,114 @@
+import logging
+from datetime import datetime, timedelta, timezone
+
+import discord
+from discord import app_commands, Interaction
+from discord.ext import commands, tasks
+
+from bot.bot import WarnetBot
+from bot.cogs.ext.temprole.time import parse_time_string
+from bot.config import GiveawayConfig, GUILD_ID
+
+logger = logging.getLogger(__name__)
+
+
+@commands.guild_only()
+class Temporary(commands.GroupCog, group_name='warnet-temp'):
+    def __init__(self, bot: WarnetBot) -> None:
+        self.bot = bot
+        self.db_pool = bot.get_db_pool()
+
+    @commands.Cog.listener()
+    async def on_connect(self) -> None:
+        self._check_temprole.start()
+
+    @app_commands.command(name='add', description='Adds a temprole to user')
+    @app_commands.describe(
+        user='User to add temprole to',
+        duration='Example: 1d2h3m4s',
+        role='Role to add',
+    )
+    async def give_role_on_poll(
+        self,
+        interaction: Interaction,
+        user: discord.Member,
+        duration: str,
+        role: discord.Role,
+    ) -> None:
+        await interaction.response.defer()
+        if role.id is GiveawayConfig.BLACKLIST_ROLE_ID:
```

Comment:
`is` is used to compare the similarity between objects. Use `==` to compare the value.